In [2]:
import pandas as pd
import numpy as np
from pathlib import Path

def _find_credit_card_csv():
    names = ("credit_card_balance.csv",)
    nested = ("credit_card_balance.csv", "credit_card_balance.csv")
    for base in [Path.cwd(), *Path.cwd().parents]:
        for data in (base / "data", base / "Proyecto_Integrador" / "data"):
            for path in (
                data / names[0],
                data.joinpath(*nested),
            ):
                if path.is_file():
                    return path
    return None

CSV_PATH = _find_credit_card_csv()
if CSV_PATH is None:
    raise FileNotFoundError(
        "No se encontró credit_card_balance.csv en data/ "
        "(revisa data/credit_card_balance.csv/credit_card_balance.csv)"
    )

cc = pd.read_csv(CSV_PATH)

print(f"Shape: {cc.shape}")
print(f"\nColumnas: {list(cc.columns)}")
print(f"\nDtypes:\n{cc.dtypes}")
print(f"\nPrimeras filas:\n{cc.head(5).to_string()}")

Shape: (3840312, 23)

Columnas: ['SK_ID_PREV', 'SK_ID_CURR', 'MONTHS_BALANCE', 'AMT_BALANCE', 'AMT_CREDIT_LIMIT_ACTUAL', 'AMT_DRAWINGS_ATM_CURRENT', 'AMT_DRAWINGS_CURRENT', 'AMT_DRAWINGS_OTHER_CURRENT', 'AMT_DRAWINGS_POS_CURRENT', 'AMT_INST_MIN_REGULARITY', 'AMT_PAYMENT_CURRENT', 'AMT_PAYMENT_TOTAL_CURRENT', 'AMT_RECEIVABLE_PRINCIPAL', 'AMT_RECIVABLE', 'AMT_TOTAL_RECEIVABLE', 'CNT_DRAWINGS_ATM_CURRENT', 'CNT_DRAWINGS_CURRENT', 'CNT_DRAWINGS_OTHER_CURRENT', 'CNT_DRAWINGS_POS_CURRENT', 'CNT_INSTALMENT_MATURE_CUM', 'NAME_CONTRACT_STATUS', 'SK_DPD', 'SK_DPD_DEF']

Dtypes:
SK_ID_PREV                      int64
SK_ID_CURR                      int64
MONTHS_BALANCE                  int64
AMT_BALANCE                   float64
AMT_CREDIT_LIMIT_ACTUAL         int64
AMT_DRAWINGS_ATM_CURRENT      float64
AMT_DRAWINGS_CURRENT          float64
AMT_DRAWINGS_OTHER_CURRENT    float64
AMT_DRAWINGS_POS_CURRENT      float64
AMT_INST_MIN_REGULARITY       float64
AMT_PAYMENT_CURRENT           float64
AMT_PAY

In [3]:
cc.head()

,SK_ID_PREV,SK_ID_CURR,MONTHS_BALANCE,AMT_BALANCE,AMT_CREDIT_LIMIT_ACTUAL,AMT_DRAWINGS_ATM_CURRENT,AMT_DRAWINGS_CURRENT,AMT_DRAWINGS_OTHER_CURRENT,AMT_DRAWINGS_POS_CURRENT,AMT_INST_MIN_REGULARITY,...,AMT_RECIVABLE,AMT_TOTAL_RECEIVABLE,CNT_DRAWINGS_ATM_CURRENT,CNT_DRAWINGS_CURRENT,CNT_DRAWINGS_OTHER_CURRENT,CNT_DRAWINGS_POS_CURRENT,CNT_INSTALMENT_MATURE_CUM,NAME_CONTRACT_STATUS,SK_DPD,SK_DPD_DEF
0,2562384,378907,-6,56.970,135000,0.0,877.5,0.0,877.5,1700.325,...,0.000,0.000,0.0,1,0.0,1.0,35.0,Active,0,0
1,2582071,363914,-1,63975.555,45000,2250.0,2250.0,0.0,0.0,2250.000,...,64875.555,64875.555,1.0,1,0.0,0.0,69.0,Active,0,0
2,1740877,371185,-7,31815.225,450000,0.0,0.0,0.0,0.0,2250.000,...,31460.085,31460.085,0.0,0,0.0,0.0,30.0,Active,0,0
3,1389973,337855,-4,236572.110,225000,2250.0,2250.0,0.0,0.0,11795.760,...,233048.970,233048.970,1.0,1,0.0,0.0,10.0,Active,0,0
4,1891521,126868,-1,453919.455,450000,0.0,11547.0,0.0,11547.0,22924.890,...,453919.455,453919.455,0.0,1,0.0,1.0,101.0,Active,0,0


| Columna | Descripción | Explicación sencilla |
|---|---|---|
| `SK_ID_PREV` | ID del crédito anterior en Home Credit | Identifica una tarjeta o crédito previo específico. Sirve para relacionar esta tabla con otras de créditos anteriores. |
| `SK_ID_CURR` | ID del cliente o solicitud actual | Es el identificador del cliente. Sirve para unir luego con `application_train`. |
| `MONTHS_BALANCE` | Mes relativo a la solicitud actual | Indica qué tan antiguo es el registro mensual. Por ejemplo, `-1` es el mes más reciente, `-2` dos meses antes, etc. |
| `AMT_BALANCE` | Balance durante el mes del crédito anterior | Saldo de la tarjeta en ese mes. Representa cuánto saldo había pendiente o usado. |
| `AMT_CREDIT_LIMIT_ACTUAL` | Límite actual de la tarjeta durante ese mes | Cupo o límite de crédito disponible/asignado para la tarjeta en ese mes. |
| `AMT_DRAWINGS_ATM_CURRENT` | Monto retirado en cajero durante ese mes | Dinero retirado en efectivo desde ATM usando la tarjeta. |
| `AMT_DRAWINGS_CURRENT` | Monto total usado durante ese mes | Total de retiros/usos de la tarjeta durante el mes. Puede incluir compras, retiros u otros movimientos. |
| `AMT_DRAWINGS_OTHER_CURRENT` | Monto de otros retiros/usos durante ese mes | Usos de la tarjeta que no son ATM ni compras POS. |
| `AMT_DRAWINGS_POS_CURRENT` | Monto usado en compras durante ese mes | Compras o consumos realizados con la tarjeta en puntos de venta. |
| `AMT_INST_MIN_REGULARITY` | Cuota mínima del mes | Pago mínimo esperado o requerido para ese mes. |
| `AMT_PAYMENT_CURRENT` | Pago realizado durante el mes | Cuánto pagó el cliente ese mes por la tarjeta. |
| `AMT_PAYMENT_TOTAL_CURRENT` | Pago total realizado durante el mes | Total pagado durante ese mes. Puede incluir distintos componentes del pago. |
| `AMT_RECEIVABLE_PRINCIPAL` | Monto por cobrar asociado al principal | Parte del saldo pendiente que corresponde al capital/principal. |
| `AMT_RECIVABLE` | Monto por cobrar | Monto que todavía se espera recibir/pagar. En el dataset aparece escrito como `RECIVABLE`, aunque parece un error ortográfico. |
| `AMT_TOTAL_RECEIVABLE` | Total por cobrar | Total pendiente por recibir de esa tarjeta/crédito. |
| `CNT_DRAWINGS_ATM_CURRENT` | Número de retiros en ATM durante el mes | Cuántas veces retiró dinero en cajero con la tarjeta. |
| `CNT_DRAWINGS_CURRENT` | Número total de usos/retiros durante el mes | Cantidad total de movimientos de uso de la tarjeta. |
| `CNT_DRAWINGS_OTHER_CURRENT` | Número de otros usos/retiros | Cantidad de usos que no fueron ATM ni POS. |
| `CNT_DRAWINGS_POS_CURRENT` | Número de compras POS | Cantidad de compras realizadas con la tarjeta en puntos de venta. |
| `CNT_INSTALMENT_MATURE_CUM` | Número acumulado de cuotas pagadas | Cuántas cuotas maduras/pagadas acumula el crédito o tarjeta. |
| `NAME_CONTRACT_STATUS` | Estado del contrato en ese mes | Indica si el contrato estaba activo, firmado, cerrado, etc. |
| `SK_DPD` | Días de mora durante ese mes | Días vencidos o en mora de la tarjeta durante ese mes. |
| `SK_DPD_DEF` | Días de mora con tolerancia | Similar a `SK_DPD`, pero aplicando tolerancia: ignora deudas pequeñas o casos menos graves. |

In [4]:
null_pct = (cc.isnull().sum() / len(cc) * 100).sort_values(ascending=False)
print(null_pct[null_pct > 0].round(2))

# Verificar si los nulos son exactamente los mismos registros
mask = cc['AMT_DRAWINGS_ATM_CURRENT'].isnull()
print(f"\nATM nulo y POS también nulo: {(mask & cc['AMT_DRAWINGS_POS_CURRENT'].isnull()).mean():.4f}")

AMT_PAYMENT_CURRENT           20.00
CNT_DRAWINGS_POS_CURRENT      19.52
AMT_DRAWINGS_ATM_CURRENT      19.52
CNT_DRAWINGS_ATM_CURRENT      19.52
AMT_DRAWINGS_POS_CURRENT      19.52
AMT_DRAWINGS_OTHER_CURRENT    19.52
CNT_DRAWINGS_OTHER_CURRENT    19.52
CNT_INSTALMENT_MATURE_CUM      7.95
AMT_INST_MIN_REGULARITY        7.95
dtype: float64

ATM nulo y POS también nulo: 0.1952


In [5]:
# Las tres columnas receivable son casi idénticas
rec_cols = ['AMT_RECEIVABLE_PRINCIPAL', 'AMT_RECIVABLE', 'AMT_TOTAL_RECEIVABLE']
print(cc[rec_cols].corr().round(4))
print(f"\nAMT_RECIVABLE == AMT_TOTAL_RECEIVABLE: {(cc['AMT_RECIVABLE'] == cc['AMT_TOTAL_RECEIVABLE']).mean():.4f}")

                          AMT_RECEIVABLE_PRINCIPAL  AMT_RECIVABLE  \
AMT_RECEIVABLE_PRINCIPAL                    1.0000         0.9997   
AMT_RECIVABLE                               0.9997         1.0000   
AMT_TOTAL_RECEIVABLE                        0.9997         1.0000   

                          AMT_TOTAL_RECEIVABLE  
AMT_RECEIVABLE_PRINCIPAL                0.9997  
AMT_RECIVABLE                           1.0000  
AMT_TOTAL_RECEIVABLE                    1.0000  

AMT_RECIVABLE == AMT_TOTAL_RECEIVABLE: 0.9971


# Eliminación de columnas redundantes
#### Eliminamos AMT_RECEIVABLE_PRINCIPAL y AMT_RECIVABLE

In [6]:
cc = cc.drop(columns=[
    "AMT_RECEIVABLE_PRINCIPAL",
    "AMT_RECIVABLE"
])

## Valores negativos en montos

In [7]:
amt_cols = [c for c in cc.columns if 'AMT' in c]
for c in amt_cols:
    neg = (cc[c] < 0).sum()
    if neg > 0:
        print(f"{c}: {neg} valores negativos ({neg/len(cc)*100:.3f}%)")

AMT_BALANCE: 2345 valores negativos (0.061%)
AMT_DRAWINGS_ATM_CURRENT: 1 valores negativos (0.000%)
AMT_DRAWINGS_CURRENT: 3 valores negativos (0.000%)
AMT_TOTAL_RECEIVABLE: 109330 valores negativos (2.847%)


In [10]:
import pandas as pd

# ==============================
# 1. Resumen de estados del contrato
# ==============================

total_filas = len(cc)

tabla_status = (
    cc["NAME_CONTRACT_STATUS"]
    .value_counts()
    .reset_index()
)

tabla_status.columns = ["estado_contrato", "cantidad"]

tabla_status["porcentaje"] = (
    tabla_status["cantidad"] / total_filas * 100
).round(2)

tabla_status

,estado_contrato,cantidad,porcentaje
0,Active,3698436,96.31
1,Completed,128918,3.36
2,Signed,11058,0.29
3,Demand,1365,0.04
4,Sent proposal,513,0.01
5,Refused,17,0.00
6,Approved,5,0.00


In [11]:
# ==============================
# 3. Estadísticos de SK_DPD
# ==============================

resumen_sk_dpd = cc["SK_DPD"].describe().reset_index()
resumen_sk_dpd.columns = ["estadístico", "valor"]

resumen_sk_dpd["valor"] = resumen_sk_dpd["valor"].round(2)

resumen_sk_dpd

,estadístico,valor
0,count,3840312.00
1,mean,9.28
2,std,97.52
3,min,0.00
4,25%,0.00
5,50%,0.00
6,75%,0.00
7,max,3260.00


La mayoría de los registros de credit_card_balance se encuentran en estado Active, lo que indica que la tabla contiene principalmente observaciones mensuales de tarjetas activas. Los estados diferentes a Active tienen una participación muy baja, especialmente Demand, Sent proposal, Refused y Approved.

En cuanto a la mora, solo el 3.99% de los registros presentan SK_DPD > 0, es decir, días de mora sin aplicar tolerancia. Al usar SK_DPD_DEF, que considera una definición más estricta o tolerante de mora, el porcentaje baja a 2.33%. Esto indica que la mayoría de los registros no presentan mora, pero los casos con mora pueden ser relevantes como señal de riesgo.

La distribución de SK_DPD está fuertemente concentrada en cero, ya que la mediana y el percentil 75 son iguales a 0. Sin embargo, el valor máximo alcanza 3260 días, lo que sugiere la presencia de casos extremos de mora que deben analizarse con cuidado antes del modelado.